packages/preview-middleware-client/src/adp/quick-actions/fe-v2/lr-enable-semantic-date-range-filter-bar.ts

```diff
@@ -33,8 +33,19 @@ export class ToggleSemanticDateRangeFilterBar
             const isActionApplicable = pageHasControlId(this.context.view, control.controlId);
             const modifiedControl = getControlById<FilterBar>(control.controlId);
             if (isActionApplicable && modifiedControl) {
-                this.isUseDateRangeTypeEnabled = modifiedControl.getProperty('useDateRangeType');
                 this.control = modifiedControl;
+
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                const id: string = this.control.getProperty('persistencyKey') ?? this.control.getId();
+                if (typeof id !== 'string') {
```

Comment:
But if you still check the type of `id` afterwards, typecasting it to `string` seems to be wrong 🤔 Shouldn't the type rather be `unknown` then? Would also solve the eslint `any` issue (I guess)

Reply:
Yes, just type casting to `string` is not really safe. There have been quite a few cases that the assumption turns out not to be correct, so I prefer to do the explicit checks. I'm not too happy about the comment to ignore the ESLint rule, I'll see if we can somehow get rid of it.

Reply:
Fixed by https://github.com/SAP/open-ux-tools/pull/2722/commits/c6df94041ce1edd6d03041b431a3ede48f77d67d